packages/types/workflow/index.ts

```diff
@@ -0,0 +1,9 @@
+import Step from '../step';
+/**
+ * Workflow is the definition holder for a set of Steps that can become a Task
+ */
+interface Workflow {
+  id: string;
+  steps: Step[];
```

Comment:
We can implement all kinds of flow control using specialized Steps.

In essence there's only branching and looping, so a `CaseStep` that could choose a next step and a `LoopStep` that wraps a step with a number of required serial/parallel executions.

There could also be a `WorkflowStep` that consists of an entire sub-workflow.